hack/e2e/run-rp-and-e2e.sh

```diff
@@ -91,25 +98,39 @@ register_sub() {
 
 run_e2e() {
     export RESOURCEGROUP=$CLUSTER_RESOURCEGROUP
-    echo "########## 🚀 Create ARO Cluster $CLUSTER ##########"
+    CLUSTER_SPN_ID=$(cat $CLUSTERSPN | jq -r .appId)
+    CLUSTER_SPN_SECRET=$(cat $CLUSTERSPN | jq -r .password)
+
+    echo "########## 🚀 Create ARO Cluster $CLUSTER - Using client-id : $CLUSTER_SPN_ID ##########"
     az aro create \
       -g "$RESOURCEGROUP" \
       -n "$CLUSTER" \
       --vnet dev-vnet \
       --master-subnet "$CLUSTER-master" \
-      --worker-subnet "$CLUSTER-worker"
+      --worker-subnet "$CLUSTER-worker" \
+      --client-id $CLUSTER_SPN_ID \
+      --client-secret $CLUSTER_SPN_SECRET
 
     echo "########## CLI : ARO List ##########"
     az aro list -o table
     echo "########## CLI : ARO list-creds ##########"
     az aro list-credentials -g "$RESOURCEGROUP" -n "$CLUSTER"
     echo "########## Run E2E ##########"
     go run ./hack/kubeadminkubeconfig "/subscriptions/$AZURE_SUBSCRIPTION_ID/resourceGroups/$RESOURCEGROUP/providers/Microsoft.RedHatOpenShift/openShiftClusters/$CLUSTER" >$KUBECONFIG
-    make e2e
+    make e2e | tee e2eoutput.txt
+    validate_e2e
+
     echo "########## CLI : ARO delete cluster ##########"
     az aro delete -g "$RESOURCEGROUP" -n "$CLUSTER" --yes
 }
 
+validate_e2e(){
+    echo "########## Validating E2E commands output ##########"
+    if [[ "$(grep 'ERROR' e2erpoutput.txt)" || "$(grep 'Panic' e2eoutput.txt)" ]]; then
+        exit 1
+    fi
```

Comment:
if we moved the `az aro delete` to a cleanup template step which was set to run always, could we avoid doing this?

Reply:
I guess the trouble here is the scope of RESOURCEGROUP and CLUSTER.  If we can solve that then the greps go away?

I don't like the idea of persisting these snippets in files, it will become spaghetti really quickly.  Can we precalculate values of RESOURCEGROUP and CLUSTER at a top level and pass them in?